### DISCUSSION TITLE
Full CUDA support

### DISCUSSION THREAD

**Discussion by ghost** [intel/llvm#4832] [Q&A] [answered]
Hi Folks,

I love what you're doing here and would like to make use of oneAPI at my company for realtime image processing. We currently use CUDA and would like to move to a platform with less vendor lock in. When do you think Nvidia GPUs will be supported as a mainline feature in the stock compiler from intel on windows? Thanks!!! :)

**Comment by pvchupin** [comment]
@dahubley, Windows support for Nvidia has been added some time ago by Codeplay folks.
Tagging @AerialMantis 
When you said "stock compiler from intel on windows" what exact compiler you refer to?

**Reply by ghost** [reply]
My understanding was the Nvidia support was only for Linux. Is this incorrect?

**Reply by pvchupin** [reply]
Windows support was added recently. See: https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#build-dpc-toolchain-with-support-for-nvidia-cuda

**Reply by ghost** [reply]
Thanks I'll have to try that! Is there a way to integrate the llvm compiler into visual studio? Sorry for all the questions.

**Reply by pvchupin** [reply]
@steffenlarsen, @bader, @alexbatashev, @vladimirlaz, do you know any good resource for that?

**Reply by alexbatashev** [reply]
Official Microsoft documentation is a good place to start: https://docs.microsoft.com/en-us/cpp/build/clang-support-msbuild?view=msvc-170

**Comment by AerialMantis** [comment, marked as answer]
Hi @dahubley, Codeplay is supporting and maintaining the CUDA backend for Nvidia GPU support in DPC++ for Windows and Linux. You can find instructions for how to build DPC++ with the CUDA backend and how to use it, and there are some video guides on our web page (https://www.codeplay.com/solutions/oneapi/for-cuda/).

On a related note, we are also contributing to a HIP backend for AMD GPU support in DPC++, which is currently experimental and supported on Linux only. There are instructions for how to build with the HIP backend on the DPC++ getting started guide (https://intel.github.io/llvm-docs/GetStartedGuide.html#build-dpc-toolchain-with-support-for-hip-amd).

We are planning to make a binary package available in the first quarter of 2022. It won't be part of the Intel oneAPI releases but we will make it available to download via the Codeplay website and will update it on a regular basis. This binary release will also come with more documentation and guides to help you get the most out of it.

If you would like to talk to us directly about our plans or have any questions you can also email us at sycl@codeplay.com

**Reply by Michoumichmich** [reply]
That's awesome 🎉 Would you happen to you know when specialization constants and/or JIT would be supported in the CUDA backend?
